version: "3.4"

services:
  django:
    container_name: django
    env_file: .env
    user: ${CURRENT_UID}
    build:
      context: .
      args:
        - DEBUG=${DEBUG}
      target: deploy
    restart: on-failure
    volumes:
      - .:/app/
      - ./static:/app/static
      - ./media:/app/media
    ports:
      - 8000:8000
    networks:
      private_network:

  postgres:
    container_name: postgres
    env_file: .env
    image: postgres:11-alpine
    restart: on-failure
    volumes:
      - route_log_db_data:/var/lib/postgresql/data
    networks:
      private_network:

networks:
  private_network:
    ipam:
      driver: default
      config:
        - subnet: 172.30.0.0/16

volumes:
  route_log_db_data:
    external: True
